### PR TITLE
fix: generating and serving registries with port

### PR DESCRIPTION
**Pull request by frezbo**
Fix generating and serving registies having port in them. This is needed to copy and serve imagecache from a vfat filesystem.

### CODE REVIEW

File: internal/app/machined/pkg/system/services/registry/registry.go [hunk 1]
```diff
@@ -123,7 +123,9 @@ func (svc *Service) handler(w http.ResponseWriter, req *http.Request) error {
 	if p.isBlob {
 		s = &singleFileStore{root: svc.root, path: "blob"}
 	} else {
-		s = &singleFileStore{root: svc.root, path: filepath.Join("manifests", ref.Name(), "digest")}
+		refName := handleRegistryWithPort(ref, p)
```
Comment (smira) [inline internal/app/machined/pkg/system/services/registry/registry.go:126]: I wonder if we could do a simple test (cc @DmitriyMV) here in the `registry_test.go` by providing a fake FS in `testdata` with some structure, but e.g. empty files, and verifying all our conversions for `:` work correctly ?

Comment (frezbo) [reply internal/app/machined/pkg/system/services/registry/registry.go:126]: added a test

File: internal/app/machined/pkg/system/services/registry/registry_test.go [hunk 2]
```diff
@@ -145,3 +150,151 @@ func readAll(t *testing.T, resp *http.Response) string {
...
+		tag          string
+	}
+
+	blobDir := filepath.Join(cacheDir, "blob")
+	manifestsDir := filepath.Join(cacheDir, "manifests")
+
+	require.NoError(t, os.MkdirAll(blobDir, 0o755))
+	require.NoError(t, os.MkdirAll(manifestsDir, 0o755))
+
+	for _, data := range []fakeRegistryData{
```
Comment (frezbo) [inline internal/app/machined/pkg/system/services/registry/registry_test.go:197]: maybe we could actually use image cache code itself to generate the cache, though we don't have a `registry:port` registry in CI

Comment (frezbo) [reply internal/app/machined/pkg/system/services/registry/registry_test.go:197]: or we could use `registry-gcr-io.ci.svc:5000` one of these in CI

Comment (frezbo) [reply internal/app/machined/pkg/system/services/registry/registry_test.go:197]: oh actually I can just use `ghcr.io:443`

Comment (smira) [reply internal/app/machined/pkg/system/services/registry/registry_test.go:197]: yeah, that might work

**Review by smira — APPROVED** [review]

**Comment by frezbo** [comment]
now we have good test coverage

**Pull request merged**